addons/point_of_sale/models/pos_session.py

```diff
@@ -290,10 +290,7 @@ def _validate_session(self):
         # Users without any accounting rights won't be able to create the journal entry. If this
         # case, switch to sudo for creation and posting.
         sudo = False
-        if (
-            not self.env['account.move'].check_access_rights('create', raise_exception=False)
-            and self.user_has_groups('point_of_sale.group_pos_user')
-        ):
+        if self.user_has_groups('point_of_sale.group_pos_user'):
```

Comment:
```suggestion
        try:
            self._create_account_move()
        except AccessError as e:
            if self.user_has_groups('point_of_sale.group_pos_user'):
                sudo = True
                self.sudo()._create_account_move()
            else:
                raise e
```

@pimodoo I could maybe only use `sudo` for POS user if a first attempt has generated an AccessError.
What do you think?

Reply:
@kitan191 It seems good to me